src/UILayer/CQRSStoreGroup.ts

```diff
@@ -0,0 +1,258 @@
+    }
+
+    /**
+     * return the state object that merge each stores's state
+     * @returns {Object} merged state object
+     * @public
+     */
+    getState<T>(): T {
+        if (this.state === InitialState) {
+            this.state = this.collectState<T>(emptyPayload);
+            return this.state;
+        }
+        return this.state as T;
+    }
+
+    // actually getState
+    private collectState<T>(payload: Payload): T {
+        const mapStateOfStore = (store: Store) => {
+            // 1. get prev or empty object
+            const prevState = this._stateCacheMap.get(store) || {};
+            const nextState = store.getState<typeof prevState>(prevState, payload);
+            if (process.env.NODE_ENV !== "production") {
+                assert.ok(typeof nextState == "object", `${store}: ${store.name}.getState() should return Object.
+e.g.)
+
+ class ExampleStore extends Store {
+     getState(prevState) {
+         return {
+            StateName: state
+         };
+     }
+ }
+ 
+Then, use can access by StateName.
+
+StoreGroup#getState()["StateName"]; // state
+
+`);
+                // Check immutability of Store'state
+                // https://github.com/almin/almin/issues/151
+                const isChangingStore = this.changingStores.indexOf(store) !== -1;
+                if (isChangingStore) {
+                    const isStateChangedAtLeastOne = Object.keys(nextState).some(key => {
+                        const prevStateValue = this.state[key];
+                        const nextStateValue = nextState[key];
+                        return prevStateValue !== nextStateValue;
+                    });
+                    assert.ok(isStateChangedAtLeastOne, `Store(${store.name}) called emitChange(). 
+But, this store's state is not changed.
+Store's state should be immutable value.`);
+                }
+            }
+            // if the prev/next state is same, not update the state.
+            if (!store.shouldStateUpdate(prevState, nextState)) {
+                return prevState;
+            }
+            // 2. update prev state. It means that update the state of the store
+            this._stateCacheMap.set(store, nextState);
+            return nextState;
+        };
+        const stateMap = this.stores.map(mapStateOfStore);
+        return Object.assign({}, ...stateMap);
+    }
+
+    /**
+     * Use `shouldStoreChange()` to let StoreGroup know if a event is not affected.
+     * The default behavior is to emitChange on every life-cycle change,
+     * and in the vast majority of cases you should rely on the default behavior.
+     *
+     * ## Example
+     *
+     * If you want to emitChange when actually changingStore that called `Store#emitChange` is larger than 0.
+     *
+     * ```js
+     * shouldStoreChange(nextState) {
+     *    return changingStores.length > 0;
+     * }
+     * ```
+     */
+    shouldStoreGroupUpdate(nextState: any): boolean {
```

Comment:
Shoule we rename `shouldStoreGroupUpdate` to `shouldStateUpdate`.

I have a concerns about `StoreLike`.
`StoreGroup#getState` and `Store#getState` is not same interface.